docs/concepts/realms.md

```diff
@@ -3,47 +3,194 @@ id: realms
+this by calling any of the possible messages in Gno.land, such as 
+[Call](../gno-tooling/cli/gnokey.md#call),
+[AddPackage](../gno-tooling/cli/gnokey.md#addpkg),
+[Send](../gno-tooling/cli/gnokey.md#send), or Run.
 
-* A unit that contains functionalities and utilities that can be used in realms.
-* Packages are stateless.
-* The default import path is `gno.land/p/~~~`.
-* Can be imported to other realms or packages.
-* Cannot import realms.
+## Working with realms
+In Gno, each transaction consists of a call stack, which is made up of `frames`.
+A single frame is a unique realm in the call stack. Every frame and its properties 
+can be accessed via different functions defined in the `std` package in Gno:
+- `std.GetOrigCaller()` - returns the address of the original signer of the
+transaction
+- `std.PrevRealm()` - returns the previous realm instance, which can be a user realm
+or a smart contract realm
+- `std.CurrentRealm()` - returns the instance of the realm that has called it
+- `std.CurrentRealmPath()` - shorthand for `std.CurrentRealm().PkgPath()`
+- `std.GetCallerAt()` - returns the n-th caller's address, going back in
+the transaction call stack
 
-#### [**Realms**](https://github.com/gnolang/gno/tree/master/examples/gno.land/r)
+Let's look at return values of these functions in two distinct situations:
+1. EOA calling a realm
+2. EOA calling a sequence of realms
 
-* Smart contracts in Gno.
-* Realms are stateful.
-* Realms can own assets (tokens).
-* The default import path is `gno.land/r/~~~`.
-* Realms can implement `Render(path string) string` to simplify dapp frontend development by allowing users to request
-  markdown renderings from validators and full nodes without a transaction.
+### 1. EOA calling a realm
+Take these two actors in the call stack:
+```
+EOA:
+    addr: g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5
+    pkgPath: "" // empty as this is a user realm
+
+Realm A:
+    addr: g17m4ga9t9dxn8uf06p3cahdavzfexe33ecg8v2s
+    pkgPath: gno.land/p/demo/users
+    
+        ┌─────────────────────┐      ┌─────────────────────────┐
+        │         EOA         │      │         Realm A         │
+        │                     │      │                         │
+        │  addr:              │      │  addr:                  │
+        │  g1jg...sqf5        ├──────►  g17m...8v2s            │
+        │                     │      │                         │
+        │  pkgPath:           │      │  pkgPath:               │
+        │  ""                 │      │  gno.land/p/demo/users  │
+        └─────────────────────┘      └─────────────────────────┘
+```
+
+Let's look at return values for each of the methods:
+```go
+std.GetOrigCaller() => `g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5`
+std.PrevRealm() => `g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5`
+std.CurrentRealm() => Realm {
+    addr:    `g17m4ga9t9dxn8uf06p3cahdavzfexe33ecg8v2s`
+    pkgPath: `gno.land/r/demo/users`
+}
+std.CurrentRealmPath() => `gno.land/r/demo/users`
```

Comment:
this one should be deprecated, do we alreacy have an issue?

Reply:
We do not have an issue. I will try to compile 2 or 3 naming / std issues that we have into one we can put on the test4 milestone board.